### DISCUSSION TITLE
Why in the nowinandroid repo do you change to io thread if room/retrofit handles things in a main safe fashion?

### DISCUSSION THREAD

**Discussion by adrem7** [android/nowinandroid#818] [Q&A] [answered]
In codelabs https://developer.android.com/codelabs/kotlin-coroutines#8 it says:

> Both Room and Retrofit make suspending functions main-safe.
> 
> It's safe to call these suspend funs from Dispatchers.Main, even though they fetch from the network and write to the database.

In the repo here we have the following code:

> class DefaultRecentSearchRepository @Inject constructor(
>     private val recentSearchQueryDao: RecentSearchQueryDao,
>     @Dispatcher(IO) private val ioDispatcher: CoroutineDispatcher,
> ) : RecentSearchRepository {
>     override suspend fun insertOrReplaceRecentSearch(searchQuery: String) {
>         withContext(ioDispatcher) {
>             recentSearchQueryDao.insertOrReplaceRecentSearchQuery(
>                 RecentSearchQueryEntity(
>                     query = searchQuery,
>                     queriedDate = Clock.System.now(),
>                 ),
>             )
>         }
>     }

Is there a reason for the ioDispatcher being provided here? Is it required? Would it be required if you were interacting with retrofit here instead of Room?

I'm looking for best practice and this info seems conflicting. Any light that can be shed on this would be great. TIA.

**Comment by SimonMarquis** [comment, marked as answer]
@adrem7 I've created a PR to remove this unnecessary dispatcher:
- #925

Don't forget to close/resolve this discussion if that's ok for you 👍

**Comment by dturner** [comment]
Agree that this dispatcher is unnecessary in this case. More information about when to use `withContext` inside repositories can be found here: https://developer.android.com/codelabs/building-a-data-layer#6
